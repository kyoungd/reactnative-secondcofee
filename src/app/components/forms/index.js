import FormHeading from './FormHeading'
import SignInForm from './SignInForm'
import SignUpForm from './SignUpForm'
import ResetPasswordForm from './ResetPasswordForm'
import PersonalInformationForm from './PersonalInformationForm'
import ProfessionalExperienceForm from './ProfessionalExperienceForm'

export {
  FormHeading,
  SignInForm,
  SignUpForm,
  ResetPasswordForm,
  PersonalInformationForm,
  ProfessionalExperienceForm
}
