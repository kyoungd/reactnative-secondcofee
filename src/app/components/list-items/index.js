import FilterOptionListItem from './FilterOptionListItem'
import JobListItem from './JobListItem'
import ProfileDetailListItem from './ProfileDetailListItem'
import SavedJobListItem from './SavedJobListItem'
import SettingsListItem from './SettingsListItem'

export {
  FilterOptionListItem,
  JobListItem,
  ProfileDetailListItem,
  SavedJobListItem,
  SettingsListItem
}
