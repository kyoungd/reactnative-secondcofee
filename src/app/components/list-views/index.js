import FilterOptionsListView from './FilterOptionsListView'
import JobsListView from './JobsListView'
import SavedJobsListView from './SavedJobsListView'

export { FilterOptionsListView, JobsListView, SavedJobsListView }
