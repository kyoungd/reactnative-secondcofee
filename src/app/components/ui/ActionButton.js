import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components/native'

const ButtonWrapper = styled.TouchableOpacity`
  height: 30;
  width: 80;
  justify-content: center;
  align-items: center;
  padding: 4px 8px;
  border-radius: 4;
  background-color: #346df1;
`

const ButtonText = styled.Text.attrs({
  numberOfLines: 1
})`
  font-family: 'SourceSansProRegular';
  font-size: 13;
  color: #fff;
  text-transform: uppercase;
`

export default function ActionButton({ text, onPress }) {
  return (
    <ButtonWrapper onPress={() => onPress()}>
      <ButtonText>{text}</ButtonText>
    </ButtonWrapper>
  )
}

ActionButton.propTypes = {
  text: PropTypes.string.isRequired,
  onPress: PropTypes.func.isRequired
}
