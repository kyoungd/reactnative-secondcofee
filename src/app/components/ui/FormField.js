import React from 'react'
import PropTypes from 'prop-types'
import { Item, Input } from 'native-base'
import styled from 'styled-components/native'

const Wrapper = styled.View`
  margin: ${props => (props.regular === true ? '0px' : '8px 0px')};
`

const FormFieldWrapper = styled(Item)`
  border-color: #dce1e7;
  background-color: ${props =>
    props.regular === true ? '#f4f6f9' : 'transparent'};
  padding: ${props => (props.regular === true ? '0px 8px' : '0px')};
  margin: 0px;
`

const TextBox = styled(Input)`
  font-family: 'SourceSansProRegular';
  font-size: 15;
  color: ${props => (props.disabled === true ? '#a5a5a5' : '#000')};
`

const Error = styled.Text`
  font-family: 'SourceSansProRegular';
  font-size: 15;
  color: #fc1c1c;
  margin: 2px 2px;
`

export default function FormField({
  label,
  value,
  error,
  secure,
  disabled,
  regular,
  keyboardType,
  returnKeyType,
  onChangeText,
  onSubmitEditing
}) {
  return (
    <Wrapper regular={regular}>
      <FormFieldWrapper regular={regular}>
        <TextBox
          disabled={disabled}
          placeholder={label}
          placeholderTextColor="#a5a5a5"
          value={value}
          secureTextEntry={secure}
          keyboardType={keyboardType}
          returnKeyType={returnKeyType}
          onChangeText={text => onChangeText(text)}
          onSubmitEditing={() => onSubmitEditing()}
        />
      </FormFieldWrapper>

      {!!error && <Error>{error}</Error>}
    </Wrapper>
  )
}

FormField.propTypes = {
  label: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  error: PropTypes.string,
  secure: PropTypes.bool,
  disabled: PropTypes.bool,
  regular: PropTypes.bool,
  keyboardType: PropTypes.string,
  returnKeyType: PropTypes.string,
  onChangeText: PropTypes.func.isRequired,
  onSubmitEditing: PropTypes.func
}
FormField.defaultProps = {
  secure: false,
  disabled: false,
  regular: false,
  keyboardType: 'default',
  returnKeyType: 'next',
  error: '',
  onSubmitEditing: () => null
}
