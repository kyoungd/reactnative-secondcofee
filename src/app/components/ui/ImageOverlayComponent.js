import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components/native'

const Overlay = styled.View`
  flex: 1;
  justify-content: center;
  padding: 16px;
  background-color: rgba(0, 0, 0, 0.8);
`

export default function ImageOverlayComponent({ children }) {
  return <Overlay>{children}</Overlay>
}

ImageOverlayComponent.propTypes = {
  children: PropTypes.node.isRequired
}
