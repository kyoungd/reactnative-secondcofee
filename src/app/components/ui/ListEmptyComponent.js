import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components/native'

const Wrapper = styled.View`
  flex: 1;
  justify-content: center;
  align-items: center;
  padding: 16px;
`

const HelpText = styled.Text`
  width: 250;
  font-family: 'SourceSansProLight';
  font-size: 20;
  text-align: center;
  color: #242635;
`

export default function ListEmptyComponent({ text }) {
  return (
    <Wrapper>
      <HelpText>{text}</HelpText>
    </Wrapper>
  )
}

ListEmptyComponent.propTypes = {
  text: PropTypes.string.isRequired
}
