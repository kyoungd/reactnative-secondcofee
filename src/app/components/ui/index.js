import CustomActivityIndicator from './CustomActivityIndicator'
import CustomStatusBar from './CustomStatusBar'
import FormField from './FormField'
import RegularButton from './RegularButton'
import ActionButton from './ActionButton'
import SafeAreaViewWrapper from './SafeAreaViewWrapper'
import Separator from './Separator'
import ListEmptyComponent from './ListEmptyComponent'
import ListLoadingComponent from './ListLoadingComponent'
import MultiSliderComponent from './MultiSliderComponent'
import ImageProgressComponent from './ImageProgressComponent'
import ImageOverlayComponent from './ImageOverlayComponent'

export {
  CustomActivityIndicator,
  CustomStatusBar,
  FormField,
  RegularButton,
  ActionButton,
  SafeAreaViewWrapper,
  Separator,
  ListEmptyComponent,
  ListLoadingComponent,
  MultiSliderComponent,
  ImageProgressComponent,
  ImageOverlayComponent
}
