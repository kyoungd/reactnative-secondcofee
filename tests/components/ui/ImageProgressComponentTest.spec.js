import React from 'react'
import { shallow } from 'react-native-testing-library'
import { ImageProgressComponent } from 'components/ui'

test('Component renders correctly', () => {
  const { output } = shallow(
    <ImageProgressComponent
      photoURL="https://imgur.com/5WzWMpx.png"
      resizeMode="cover"
      style={{ width: 80, height: 80 }}
      imageStyle={{ borderRadius: 40 }}
    />
  )
  expect(output).toMatchSnapshot()
})
