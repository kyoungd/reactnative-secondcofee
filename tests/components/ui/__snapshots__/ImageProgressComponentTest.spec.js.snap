// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component renders correctly 1`] = `
<ForwardRef
  imageStyle={
    Object {
      "borderRadius": 40,
    }
  }
  resizeMode="cover"
  source={
    Object {
      "uri": "https://imgur.com/5WzWMpx.png",
    }
  }
  style={
    Object {
      "height": 80,
      "width": 80,
    }
  }
/>
`;
